[JIT] OpInfo tests for nvfuser

davidberard98 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #75116
* __->__ #71299
* #73322
* #74361

These tests verify that for the same inputs, the eager version of an op
and a traced, fused version of the op return the same output.

Currently the tests don't check whether or not fusion actually occurred.

Differential Revision: [D33595299](https://our.internmc.facebook.com/intern/diff/D33595299)

davidberard98 committed:
[WIP][JIT] OpInfo tests for nvfuser

These tests verify that for the same inputs, the eager version of an op
and a traced, fused version of the op return the same output.

Currently the tests don't check whether or not fusion actually occurred.

[ghstack-poisoned]

davidberard98 committed:
Update on "[WIP][JIT] OpInfo tests for nvfuser"


These tests verify that for the same inputs, the eager version of an op
and a traced, fused version of the op return the same output.

Currently the tests don't check whether or not fusion actually occurred.

[ghstack-poisoned]

davidberard98 committed:
Update on "[WIP][JIT] OpInfo tests for nvfuser"


These tests verify that for the same inputs, the eager version of an op
and a traced, fused version of the op return the same output.

Currently the tests don't check whether or not fusion actually occurred.

[ghstack-poisoned]

davidberard98 committed:
Update on "[WIP][JIT] OpInfo tests for nvfuser"


These tests verify that for the same inputs, the eager version of an op
and a traced, fused version of the op return the same output.

Currently the tests don't check whether or not fusion actually occurred.

[ghstack-poisoned]

davidberard98 commented:
current test failures: [nvfuser-opinfo.txt](https://gist.github.com/davidberard98/e7065260fb3b7549879ec17904d0cd84)

davidberard98 committed:
Update on "[WIP][JIT] OpInfo tests for nvfuser"


These tests verify that for the same inputs, the eager version of an op
and a traced, fused version of the op return the same output.

Currently the tests don't check whether or not fusion actually occurred.

[ghstack-poisoned]

eellison commented on test/test_jit_cuda_fuser.py:
```diff
@@ -3198,6 +3204,33 @@ def test_register_fuser(self):
 
+_tracing_ops = partial(ops, dtypes=OpDTypes.supported,
+                       allowed_dtypes=(torch.float, torch.cfloat))
+
```
is there a reason we're restricting to float and cfloat ?

davidberard98 replied:
copied this from the [variant_consistency tests](https://github.com/pytorch/pytorch/blob/master/test/test_ops.py#L32), where 

```python
# variant testing is only done with torch.float and torch.cfloat to avoid
#   excessive test times and maximize signal to noise ratio
```

eellison replied:
there's a slow test that runs nightly, we could at least run it then (and initially, to flush out issue)

eellison replied:
ctrl-f SLOW_TEST or something and you'll find it

davidberard98 committed:
Update on "[WIP][JIT] OpInfo tests for nvfuser"


These tests verify that for the same inputs, the eager version of an op
and a traced, fused version of the op return the same output.

Currently the tests don't check whether or not fusion actually occurred.

[ghstack-poisoned]

davidberard98 committed:
Update on "[WIP][JIT] OpInfo tests for nvfuser"


These tests verify that for the same inputs, the eager version of an op
and a traced, fused version of the op return the same output.

Currently the tests don't check whether or not fusion actually occurred.

[ghstack-poisoned]

davidberard98 committed:
Update on "[WIP][JIT] OpInfo tests for nvfuser"


These tests verify that for the same inputs, the eager version of an op
and a traced, fused version of the op return the same output.

Currently the tests don't check whether or not fusion actually occurred.

[ghstack-poisoned]

davidberard98 committed:
Update on "[WIP][JIT] OpInfo tests for nvfuser"


These tests verify that for the same inputs, the eager version of an op
and a traced, fused version of the op return the same output.

Currently the tests don't check whether or not fusion actually occurred.

[ghstack-poisoned]

davidberard98 commented:
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33595299).

davidberard98 committed:
Update on "[WIP][JIT] OpInfo tests for nvfuser"


These tests verify that for the same inputs, the eager version of an op
and a traced, fused version of the op return the same output.

Currently the tests don't check whether or not fusion actually occurred.

Differential Revision: [D33595299](https://our.internmc.facebook.com/intern/diff/D33595299)

[ghstack-poisoned]

davidberard98 commented:
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33595299).

davidberard98 committed:
Update on "[WIP][JIT] OpInfo tests for nvfuser"


These tests verify that for the same inputs, the eager version of an op
and a traced, fused version of the op return the same output.

Currently the tests don't check whether or not fusion actually occurred.

Differential Revision: [D33595299](https://our.internmc.facebook.com/intern/diff/D33595299)

[ghstack-poisoned]

davidberard98 commented:
updated [current results.txt](https://gist.github.com/davidberard98/c6c14c786e5d0e90f6efa38ed81df4a7). Previous iteration wasn't actually calling nvfuser...

davidberard98 committed:
Update on "[WIP][JIT] OpInfo tests for nvfuser"


These tests verify that for the same inputs, the eager version of an op
and a traced, fused version of the op return the same output.

Currently the tests don't check whether or not fusion actually occurred.

Differential Revision: [D33595299](https://our.internmc.facebook.com/intern/diff/D33595299)

[ghstack-poisoned]

jjsjann123 commented:
Looks like a lot of the failure comes from 0-dim cuda tensors. :stuck_out_tongue: 
We had our scalar tensor / 0-dim empty buffer support mixed up in integration and have been trying to get a refactor. That keeps getting kicked further down the road (totally my fault :stuck_out_tongue: ), but I think it's finally the time.

I'm jumping on that. But likely we won't be putting that in current upstream PR. I'll try to wrap thing up and push another upstream PR once the existing one is merged.

davidberard98 committed:
Update on "[WIP][JIT] OpInfo tests for nvfuser"


These tests verify that for the same inputs, the eager version of an op
and a traced, fused version of the op return the same output.

Currently the tests don't check whether or not fusion actually occurred.

Differential Revision: [D33595299](https://our.internmc.facebook.com/intern/diff/D33595299)

[ghstack-poisoned]

davidberard98 committed:
Update on "[WIP][JIT] OpInfo tests for nvfuser"


These tests verify that for the same inputs, the eager version of an op
and a traced, fused version of the op return the same output.

Currently the tests don't check whether or not fusion actually occurred.

Differential Revision: [D33595299](https://our.internmc.facebook.com/intern/diff/D33595299)

[ghstack-poisoned]

davidberard98 committed:
Update on "[WIP][JIT] OpInfo tests for nvfuser"


These tests verify that for the same inputs, the eager version of an op
and a traced, fused version of the op return the same output.

Currently the tests don't check whether or not fusion actually occurred.

Differential Revision: [D33595299](https://our.internmc.facebook.com/intern/diff/D33595299)

[ghstack-poisoned]

davidberard98 committed:
Update on "[WIP][JIT] OpInfo tests for nvfuser"


These tests verify that for the same inputs, the eager version of an op
and a traced, fused version of the op return the same output.

Currently the tests don't check whether or not fusion actually occurred.

Differential Revision: [D33595299](https://our.internmc.facebook.com/intern/diff/D33595299)

[ghstack-poisoned]

davidberard98 commented:
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33595299).

davidberard98 committed:
Update on "[WIP][JIT] OpInfo tests for nvfuser"


These tests verify that for the same inputs, the eager version of an op
and a traced, fused version of the op return the same output.

Currently the tests don't check whether or not fusion actually occurred.

Differential Revision: [D33595299](https://our.internmc.facebook.com/intern/diff/D33595299)

[ghstack-poisoned]

davidberard98 commented:
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33595299).

davidberard98 committed:
Update on "[WIP][JIT] OpInfo tests for nvfuser"


These tests verify that for the same inputs, the eager version of an op
and a traced, fused version of the op return the same output.

Currently the tests don't check whether or not fusion actually occurred.

Differential Revision: [D33595299](https://our.internmc.facebook.com/intern/diff/D33595299)

[ghstack-poisoned]

davidberard98 commented:
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33595299).

davidberard98 committed:
Update on "[JIT] OpInfo tests for nvfuser"


These tests verify that for the same inputs, the eager version of an op
and a traced, fused version of the op return the same output.

Currently the tests don't check whether or not fusion actually occurred.

Differential Revision: [D33595299](https://our.internmc.facebook.com/intern/diff/D33595299)

[ghstack-poisoned]

davidberard98 committed:
Update on "[JIT] OpInfo tests for nvfuser"


These tests verify that for the same inputs, the eager version of an op
and a traced, fused version of the op return the same output.

Currently the tests don't check whether or not fusion actually occurred.

Differential Revision: [D33595299](https://our.internmc.facebook.com/intern/diff/D33595299)

[ghstack-poisoned]

davidberard98 commented:
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33595299).

davidberard98 committed:
Update on "[JIT] OpInfo tests for nvfuser"


These tests verify that for the same inputs, the eager version of an op
and a traced, fused version of the op return the same output.

Currently the tests don't check whether or not fusion actually occurred.

Differential Revision: [D33595299](https://our.internmc.facebook.com/intern/diff/D33595299)

[ghstack-poisoned]

davidberard98 commented:
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33595299).

davidberard98 committed:
Update on "[JIT] OpInfo tests for nvfuser"


These tests verify that for the same inputs, the eager version of an op
and a traced, fused version of the op return the same output.

Currently the tests don't check whether or not fusion actually occurred.

Differential Revision: [D33595299](https://our.internmc.facebook.com/intern/diff/D33595299)

[ghstack-poisoned]

davidberard98 commented:
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33595299).

eellison commented on torch/testing/_internal/common_methods_invocations.py:
```diff
@@ -8647,6 +8647,7 @@ def ref_pairwise_distance(input1, input2):
                # https://github.com/pytorch/pytorch/issues/71784
                DecorateInfo(unittest.skip('Skipped!'), 'TestNNCOpInfo', 'test_nnc_correctness',
                             device_type='cpu', dtypes=(torch.float16,)),
+               DecorateInfo(unittest.skip('Skipped!'), 'TestCudaFuserOpInfo', 'test_nvfuser_correctness', dtypes=(torch.float16,)),
```
Cool! should we file issues for the failing tests ?

davidberard98 replied:
I think most of them either have an issue filed or are expected to fail

e.g. https://github.com/pytorch/pytorch/issues/71784 for this one

jjsjann123 replied:
Let me jump on the failing tests!

davidberard98 replied:
@jjsjann123 fyi I think #71784 might be expected

And list of tests that need fixes is in #75029 (also see this board: https://github.com/pytorch/pytorch/projects/30)

eellison reviewed (approved):
😍 😍 😍  Should we file issues for the failing tests ?
